pkg/health/server.go

```diff
@@ -77,7 +77,12 @@ func (s *server) Run(ctx context.Context, port int) error {
 		serveErr <- nil
 	}()
 
-	<-ctx.Done()
+	select {
+	case err := <-serveErr:
+		return err
```

Comment:
since serveErr is returned here, I think line 94 should be changed to return the error returned from Shutdown call. Also this call to `<-serveErr` in line 94 can block indefinitely, if for some reason `ListenAndServe` returns before the context is done?